### ISSUE TITLE
Allow FTP sources

### ISSUE THREAD

**Issue by alexlarsson** [flatpak/flatpak-builder#23]
_From @Alexander-Wilms on March 28, 2017 2:41_

I'm trying to build TeX Live, but the official archive is only available via FTP: ftp://tug.org/texlive/historic/2016/texlive-20160523b-source.tar.xz

Adding the url as a source results in the following error message: "Failed to download sources: module texlive: Not supported address schema"

_Copied from original issue: flatpak/flatpak#666_

**Comment by alexlarsson** [comment]
Yeah, libsoup (which we use for downloads) has no ftp support. We would need some nice ftp library to add this.

**Comment by alexlarsson** [comment]
In previous examples of this I have always been able to find a http mirror though.

**Comment by bbhtt** [comment]
FTP seems pretty dead to me in 2025
